CMakeLists.txt

```diff
@@ -13,12 +13,30 @@ option(BUILD_SHARED_LIBS "Build shared libraries (.dll/.so) instead of static on
 option(SENTRY_BUILD_TESTS "Build sentry-native tests" ON)
 option(SENTRY_BUILD_EXAMPLES "Build sentry-native example(s)" ON)
 
-set(SENTRY_CURL_DEFAULT ON)
 if(WIN32)
-	set(SENTRY_CURL_DEFAULT OFF)
+	set(SENTRY_DEFAULT_TRANSPORT "winhttp")
+elseif(ANDROID)
+	set(SENTRY_DEFAULT_TRANSPORT "none")
+else()
+	set(SENTRY_DEFAULT_TRANSPORT "curl")
```

Comment:
Should we swap these conditions?

- Win32 -> winhttp
- Linux / macOS -> curl
- Android -> none (explicit)
- else -> none